geostore/check_stac_metadata/task.py

```diff
@@ -64,21 +67,37 @@ def lambda_handler(event: JsonObject, _context: bytes) -> JsonObject:
 
     try:
         s3_client = get_s3_client_for_role(event[S3_ROLE_ARN_KEY])
+        geostore_s3_client = get_s3_client_for_role(get_param(ParameterName.S3_USERS_ROLE_ARN))
     except ClientError as error:
         LOGGER.warning(LOG_MESSAGE_LAMBDA_FAILURE, extra={"error": error})
         return {ERROR_MESSAGE_KEY: str(error)}
 
-    def s3_url_reader(url: str) -> StreamingBody:
+    def s3_url_reader(url: str, dataset_prefix: str) -> StreamingBody:
```

Comment:
Do we want to test this function in isolation? Then we shouldn't need more than a single test to ensure it's being used in the Lambda to ensure everything works.